lputils.pas

```diff
@@ -380,30 +381,45 @@ procedure ExposeGlobals(Compiler: TLapeCompiler; HeaderOnly, DoOverride: Boolean
       'end;';
   end;
 
+var
+  Code: lpString;
 begin
   if (Compiler = nil) then
     Exit;
 
-  Compiler.addDelayedCode(
-    LapeDelayedFlags +
-    GetGlobalPtr()  + LineEnding +
-    GetGlobalName() + LineEnding +
-    GetGlobalVal()  + LineEnding +
-    VariantInvoke() + LineEnding +
-    ToString()
-  , '!addDelayedExpose');
+  Code := LapeDelayedFlags;
+
+  if Compiler.hasDefine('Lape_VariantInvoke') then
+    Code := Code + VariantInvoke();
+  if Compiler.hasDefine('Lape_GetGlobalValue') then
+    Code := Code + GetGlobalVal();
+  if Compiler.hasDefine('Lape_GetGlobalPtr') then
+    Code := Code + GetGlobalPtr();
+  if Compiler.hasDefine('Lape_GetGlobalName')  then
+    Code := Code + GetGlobalName() + ToString();
+
+  Compiler.addDelayedCode(Code, '!addDelayedExpose');
 end;
 
 procedure _ExposeGlobals_FillProcs(Compiler: TLapeCompiler);
 begin
   ExposeGlobals(Compiler, False, True);
 end;
 
-procedure ExposeGlobals(Compiler: TLapeCompiler);
+procedure ExposeGlobals(Compiler: TLapeCompiler; Methods: EExposeGlobalsMethods);
 begin
   if (Compiler = nil) then
     Exit;
 
+  if egmInvoke in Methods then
+    Compiler.addBaseDefine('Lape_VariantInvoke');
+  if egmName in Methods then
+    Compiler.addBaseDefine('Lape_GetGlobalName');
+  if egmPtr in Methods then
+    Compiler.addBaseDefine('Lape_GetGlobalPtr');
+  if egmValue in Methods then
+    Compiler.addBaseDefine('Lape_GetGlobalValue');
+
   ExposeGlobals(Compiler, True, False);
   Compiler.addBaseDefine('Lape_ExposeGlobals');
   Compiler.AfterParsing.AddProc(@_ExposeGlobals_FillProcs);
```

Comment:
I think it may be better to split `_ExposeGlobals_FillProcs` so that we can `AfterParsing.AddProc` only the necessary filler functions.